delira/logging/base_logger.py

```diff
@@ -2,6 +2,179 @@
 from queue import Full
 from delira.logging.base_backend import BaseBackend
 import logging
+import numpy as np
+from types import FunctionType
+from collections import Iterable, MutableMapping
+
+
+# Reduction Functions
+def _reduce_last(items: list):
```

Comment:
The reduce functions are the same as defined inside the `BaseTrainer`, I think we should have one place where we define them (and the dictionary which maps the strings to the functions) and import them from there.